src/formatting_layer.rs

```diff
@@ -74,15 +76,39 @@ impl<W: for<'a> MakeWriter<'a> + 'static> BunyanFormattingLayer<W> {
         Self::with_default_fields(name, make_writer, HashMap::new())
     }
 
-    pub fn with_default_fields(name: String, make_writer: W, default_fields: HashMap<String, Value>) -> Self {
+    pub fn with_default_fields(
+        name: String,
+        make_writer: W,
+        default_fields: HashMap<String, Value>,
+    ) -> Self {
         Self {
             make_writer,
             name,
             pid: std::process::id(),
             hostname: gethostname::gethostname().to_string_lossy().into_owned(),
             bunyan_version: 0,
             default_fields,
+            skip_fields: HashSet::new(),
+        }
+    }
+
+    /// Add fields to skip when formatting with this layer.
+    pub fn skip_fields<T>(mut self, fields: &[T]) -> Result<Self, String>
+    where
+        T: AsRef<str>,
+    {
+        for field in fields {
+            let field = field.as_ref();
+            if BUNYAN_RESERVED_FIELDS.contains(&field) {
+                return Err(format!(
+                    "{} is a core field in the bunyan log format, it can't be skipped",
+                    field
+                ));
```

Comment:
Better to avoid returning a `String` as error.
Let's introduce a `InvalidFieldError` struct, marked with `#[non_exhaustive]` that implements `Display` + `Error`.

Reply:
Sure, fixed in the latest commit